slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/DbHealthCheck.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2022 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.slashingprotection;
+
+import java.util.Optional;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class DbHealthCheck implements Runnable {
+
+  private final Optional<SlashingProtectionContext> slashingProtectionContext;
+  private final int dbHealthCheckTimeoutMilliseconds;
+  private final int dbHealthCheckIntervalMilliseconds;
+  private final AtomicBoolean isDbDown;
+
+  public DbHealthCheck(
+      final Optional<SlashingProtectionContext> slashingProtectionContext,
+      final int dbHealthCheckTimeoutMilliseconds,
+      final int dbHealthCheckIntervalMilliseconds) {
+    this.slashingProtectionContext = slashingProtectionContext;
+    this.dbHealthCheckTimeoutMilliseconds = dbHealthCheckTimeoutMilliseconds;
+    this.dbHealthCheckIntervalMilliseconds = dbHealthCheckIntervalMilliseconds;
+    this.isDbDown = new AtomicBoolean(false);
+  }
+
+  @Override
+  public void run() {
+    slashingProtectionContext.ifPresent(
+        protectionContext ->
+            Executors.newScheduledThreadPool(1)
```

Comment:
We shouldn't create the thread pool each time. It would be better to create the thread pool in the runner and schedule running the DbHealthCheck there so this only need to worry about doing the health check.